core/claims/claimKeyBabyJub.go

```diff
@@ -0,0 +1,86 @@
+package claims
+
+import (
+	"encoding/binary"
+	"math/big"
+
+	"github.com/iden3/go-iden3-core/common"
+	"github.com/iden3/go-iden3-core/merkletree"
+	"github.com/iden3/go-iden3-crypto/babyjub"
+)
+
+// ClaimKeyBabyJub is a claim to authorize a baby jub public key for
+// signing.
+type ClaimKeyBabyJub struct {
+	metadata Metadata
+	// KeyType is used to specify if the type of usage given to the Key
+	// if 0 is the generic one
+	// if 1 is to AuthorizeKSign Key
+	KeyType uint64
+	// Sign means positive if false, negative if true.
+	Sign bool
+	// Ay is the y coordinate of the baby jub curve point which corresponds
+	// to the public key.
+	Ay *big.Int
+}
+
+// NewClaimKeyBabyJub returns a ClaimKeyBabyJub with the
+// given elliptic public key parameters.
+func NewClaimKeyBabyJub(pk *babyjub.PublicKey, subType uint64) *ClaimKeyBabyJub {
+	return &ClaimKeyBabyJub{
+		metadata: NewMetadata(ClaimHeaderKeyBabyJub),
+		KeyType:  subType,
+		Sign:     babyjub.PointCoordSign(pk.X),
+		Ay:       pk.Y,
+	}
+}
+
+// NewClaimKeyBabyJubFromEntry deserializes a
+// ClaimKeyBabyJubFrom from an Entry.
+func NewClaimKeyBabyJubFromEntry(e *merkletree.Entry) *ClaimKeyBabyJub {
+	c := &ClaimKeyBabyJub{}
+	c.metadata.Unmarshal(e)
+
+	c.KeyType = binary.BigEndian.Uint64(e.Data[1][:])
+
+	sign := []byte{0}
+	copy(sign, e.Data[2][:])
+	if sign[0] == 1 {
+		c.Sign = true
+	}
+	c.Ay = new(big.Int).SetBytes(common.SwapEndianness(e.Data[3][:]))
+	return c
+}
+
+// Entry serializes the claim into an Entry.
+func (c *ClaimKeyBabyJub) Entry() *merkletree.Entry {
+	e := &merkletree.Entry{}
+	index := e.Index()
+
+	var subType [8]byte
+	binary.BigEndian.PutUint64(subType[:], c.KeyType)
```

Comment:
:exclamation:

Reply:
About the `BigEndian`, I'm agree, in fact we agreed to when we end the current `phase` we will remove finally all the `BigEndians` from the code to use officially `LittleEndian`. But in this case I've putted `BigEndian` here to have it according with the ClaimType code which uses `BigEndian`: https://github.com/iden3/go-iden3-core/blob/master/core/claims/claim.go#L100

Reply:
But, as is planned, at the end of the current `phase` we will get rid off all `BigEndians` to enter a new prosper era with only `LittleEndian`